### ISSUE TITLE
False positive for  `unused_enumerated` 

### ISSUE THREAD

**Issue by carl8382** [realm/SwiftLint#5881]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

There's a false positive for the rule `unused_enumerated`. The offset given from enumerated is used, just apparently not fast enough.

```swift
// This triggers a unused_enumerated  violation:
let previousValueComponent = "1234"
let currentValueComponent = "1434"

                // Compare previous and current components and find if there are any indexes where:
                // - two characters are digits
                // - these two digits are different
                //
                let firstIndexWithDifferentDigit = zip(previousValueComponent, currentValueComponent)
                    .enumerated()
                    .first {
                        $0.element.0.isNumber &&
                        $0.element.1.isNumber &&
                        $0.element.0 != $0.element.1
                    }?.offset

                return firstIndexWithDifferentDigit
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint [--no-cache] [--fix]
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure) 0.57.1
* Xcode version (run `xcodebuild -version` to be sure) 16.0
* Installation method used (Homebrew, CocoaPods, building from source, etc) SPM
* Configuration file:

```yml
# insert yaml contents here
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

**Comment by ortekka** [comment]
For reference, here's an even simpler code snippet that triggers a false positive:

```swift
public extension Array where Element: Comparable {
	var maxIndex: Int? {
		self.enumerated().max { $0.element < $1.element }?.offset
	}
}
```
Just because `offset` isn't used in the closure doesn't mean that `.enumerated()` can be removed, as the warning message incorrectly suggests.

**Comment by rgoldberg** [comment]
Is this scheduled to be fixed? Thanks.

**Comment by SimplyDanny** [comment]
> Is this scheduled to be fixed? Thanks.

Well, we don't schedule topics. Whoever wants to, may work on it.

**Comment by theamodhshetty** [comment]
Opened #6544 for this. The fix carries result-member usage from the surrounding higher-order call back into the  check, so cases like  stop getting flagged while the existing real violations still trigger.

**Comment by theamodhshetty** [comment]
Opened #6544 for this. The fix carries result-member usage from the surrounding higher-order call back into the rule, so cases like `.max { ... }?.offset` stop getting flagged while the existing real violations still trigger.
